### PR TITLE
OPS-5898 Update Aerospike Connect for Spark 3.4.1_spark3.1

### PR DESCRIPTION
**Pull request by pvinh-spike**

### CODE REVIEW

**Comment by mcoberly2** [comment]
Do these files also need to be update to 3.4.1?

- notebooks/spark/feature-store-feature-eng.ipynb
- notebooks/spark/feature-store-model-serving.ipynb
- notebooks/spark/feature-store-model-training.ipynb

**Review by mcoberly2 — APPROVED** [review]
LGTM

**Comment by neelp-git** [comment]
I am seeing the same errors as reported by @artanderson. 

Py4JJavaError: An error occurred while calling o156.save (or load).
: java.lang.ClassNotFoundException: Failed to find data source: aerospike.

3.4.1 seems broken. @rahul-aerospike, can you shed light on whether it is backward compatible with 3.4.0 and if it was tested with any Spark notebooks? Thanks.

**Comment by rahul-aerospike** [comment]
@artanderson <https://github.com/artanderson> , @Neel Phadnis
***@***.***>, official connector scala/pyspark notebooks run
fine with 3.4.1 release binary. Just to be sure, I re-ran and re-tested
them with aerospike-spark-3.4.1_spark_3.1_clientunshaded.jar. Above stated
notebooks run fine.
I also ran the feature-store notebook separately, everything looked fine to
me. Here is the screen recording
<https://drive.google.com/drive/folders/1dos8onHSTnS7GmYJojVtKGsm7DSYRQ_2?usp=sharing>
.

Best,
Rahul

**Comment by neelp-git** [comment]
Thanks @rahul-aerospike for checking. Where are you getting the aerospike-spark-3.4.0_spark_3.1_clientunshaded.jar from? 

The official jar on the download site is invalid:
-rw-r--r-- 1 jovyan users 23541404 May 10 00:58 aerospike-spark-3.4.0_spark_3.1_clientunshaded.jar
-rw-r--r-- 1 jovyan users     4073 Jun 29 14:46 aerospike-spark-3.4.1_spark_3.1_clientunshaded.jar

**Comment by artanderson** [comment]
@rahul-aerospike, I guess I'm confused by your video... Are running the notebook locally, with edited code? The problem we are running into is that changing from 3.4.0 to 3.4.1 breaks the notebooks running in the container.

[Here is a screen recording](https://drive.google.com/file/d/11k96zKNXxZgzMpV3nplbYIHh_RjErz8H/view?usp=sharing) showing the issue. I've added both 3.4.0 and 3.4.1 to the container and as you will see, changing the version, and nothing else, breaks the notebook. What are we missing here?

**Comment by rahul-aerospike** [comment]
Yes, I was running the notebook locally. The only change I made in the
notebook was directly specifying the connector path (thus skipping it's
download process) and using my local spark setup. The reason to run locally
was to isolate any docker setup issue with the plain failure of recently
released to run.

Download link for 3.4.0 and 3.4.1 have changed
https://download.aerospike.com/artifacts/aerospike-spark/*3.4.1*
/aerospike-spark-3.4.1_spark_3.1_clientunshaded.jar
https://download.aerospike.com/artifacts/aerospike-spark/*3.4.0_spark3.1*
/aerospike-spark-3.4.0_spark_3.1_clientunshaded.jar

I tried running the notebook into a fresh ubuntu VM and I got the same
error. However If I look at the terminal logs I see
*java.io.FileNotFoundException: Jar
/home/jovyan/notebooks/spark/aerospike-spark-3.4.0_spark_3.1_clientunshaded.jar
not found*
at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:1929)
......

Can you please check your terminal log. If you're getting the same error
then it means you'll have to update the download part.  Please let me know.

Best,
Rahul

File: Dockerfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ RUN mkdir /opt/spark-nb; cd /opt/spark-nb\
   && wget -qO- "https://javadl.oracle.com/webapps/download/AutoDL?BundleId=245467_4d5417147a92418ea8b615e228bb6935" | tar -xvz \
   && wget -qO- "https://archive.apache.org/dist/spark/spark-3.1.2/spark-3.1.2-bin-hadoop3.2.tgz" | tar -xvz \
   && pip install findspark numpy pandas matplotlib sklearn \
-  && wget "https://download.aerospike.com/artifacts/aerospike-spark/3.4.0_spark3.1/aerospike-spark-3.4.0_spark_3.1_clientunshaded.jar"
+  && wget "https://download.aerospike.com/artifacts/aerospike-spark/3.4.1_spark3.1/aerospike-spark-3.4.1_spark_3.1_clientunshaded.jar"
```
Comment (neelp-git) [inline Dockerfile:28]: Should be:

&& wget "https://download.aerospike.com/artifacts/aerospike-spark/3.4.1/aerospike-spark-3.4.1_spark_3.1_clientunshaded.jar"

Comment (mcoberly2) [reply Dockerfile:28]: Agreed. I will update this.

Comment (mcoberly2) [reply Dockerfile]: Done updated.

**Review by artanderson — APPROVED** [review]
Looks good, thanks!

**Pull request merged**